src/libertem/api.py

```diff
@@ -602,7 +607,7 @@ def run(self, job: Union[Job, Analysis], roi=None):
             return analysis.get_results(out)
         return out
 
-    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False):
+    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False) -> dict:
```

Comment:
I would prefer a more concrete type here - what are the key and value types of this `dict`?